web/src/user_groups.ts

```diff
@@ -354,6 +355,14 @@ export function check_system_user_group_allowed_for_setting(
         return false;
     }
 
+    if (
+        group_name === "role:fullmembers" &&
+        for_new_settings_ui &&
+        realm.realm_waiting_period_threshold === 0
+    ) {
+        return false;
```

Comment:
We probably want a comment explaining that we hide the full members group when it's not defined as separate from members.

Reply:
Added a comment.